GVRf/Framework/src/org/gearvrf/GVRActivity.java

```diff
@@ -119,13 +120,28 @@ protected void onDestroy() {
      */
     public void setScript(GVRScript gvrScript, String distortionDataFileName) {
         if (getRequestedOrientation() == ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE) {
-            mGVRViewManager = new GVRViewManager(this, gvrScript,
-                    distortionDataFileName);
+            if(isVrSupported()) {
+                mGVRViewManager = new GVRViewManager(this, gvrScript, distortionDataFileName);
+            } else {
+                mGVRViewManager = new GVRMonoscopicViewManager(this, gvrScript, distortionDataFileName);
+            }
         } else {
             throw new IllegalArgumentException(
                     "You can not set orientation to portrait for GVRF apps.");
         }
     }
+    
+    private boolean isVrSupported() {
+        if((Build.MODEL.contains("SM-N910")) || 
```

Comment:
Total quibble, but: this could be 

    return Build.MODEL.contains("SM-N910") || ...

... or even iterate over an array of model names.